pkg/domain/infra/abi/containers.go

```diff
@@ -1539,6 +1540,11 @@ func (ic *ContainerEngine) ContainerClone(ctx context.Context, ctrCloneOpts enti
 	}
 
 	if ctrCloneOpts.CreateOpts.Pod != "" {
+		if len(spec.Networks) > 0 {
+			logrus.Warning("resetting network config, cannot specify a network other than the pod's when sharing the net namespace")
```

Comment:
You should reset network info only when the netns it is shared, see the loop below.